n_helpers.c

```diff
@@ -226,26 +226,30 @@ const char * NoteBinaryDataReset(void)
   @param  inLen The length of the binary payload.
   @param  outBuf The buffer to write the decoded payload to. This can be the
                  same address as inBuf, allowing for in-place decoding.
-  @param  outLen On input, holds the length of outBuf. On output, holds the
-                 length of the decoded data.
+  @param  outLen The length of outBuf.
+  @param  decLen [out] The length of the decoded data.
 
   @returns  NULL on success, else an error string pointer.
+
+  @note When in-place decoding, the overall length shrinks and the
+        decoded data will be left-justified in the buffer.
  */
 /**************************************************************************/
 const char * NoteBinaryDecode(const uint8_t *inBuf, uint32_t inLen,
-                              uint8_t *outBuf, uint32_t *outLen)
+                              uint8_t *outBuf, uint32_t outLen,
+                              uint32_t *decLen)
 {
-    if (inBuf == NULL || outBuf == NULL || outLen == NULL) {
+    if (inBuf == NULL || outBuf == NULL || decLen == NULL) {
         NOTE_C_LOG_ERROR("NULL parameter");
         return ERRSTR("NULL parameter", c_err);
     }
 
-    if (*outLen < cobsGuaranteedFit(inLen)) {
+    if (outLen < cobsGuaranteedFit(inLen)) {
```

Comment:
`cobsGuaranteedFit` adds the +1 for the transport framing. I think we can make things more consistent, by making a clear separation between the pure COBS encoding requirements and the transport framing requirements (the +1.). Adding `#define COBS_TRANSPORT_OVERHEAD=1` will make this easier to spot in the code.

Reply:
I like this idea. To follow the common naming convention, I will use the symbol `COBS_EOP_OVERHEAD`.

From Wikipedia:
```
   [OHB]                                : Overhead byte (Start of frame)
     3+ -------------->|                : Points to relative location of first zero symbol
                       2+-------->|     : Is a zero data byte, pointing to next zero symbol
                                  [EOP] : Location of end-of-packet zero symbol.
     0     1     2     3     4    5     : Byte Position
     03    11    22    02    33   00    : COBS Data Frame
           11    22    00    33         : Extracted Data
     
OHB = Overhead Byte (Points to next zero symbol)
EOP = End Of Packet
```